ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-templates.md

```diff
@@ -0,0 +1,64 @@
+<!--
+  ~ Copyright (c) 2023 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# Custom Templates
+
+With Custom Templates for the `eos_designs` role, the user can create additional structured configuration variables that would otherwise not be created by the builtin templates.
+These additional variables will then be available for use by other AVD roles.
+
+For example, the user may require additional structured configuration variables to render a custom template the user has created for the `eos-cli-config-gen` role.
+
+## Create a Custom Template
+
+Custom templates should be written in Jinja2.
+
+For details on how to create Jinja2 templates, see [Templating Jinja2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_templating.html#templating-jinja2)
+
+For example:
+
+```jinja2
+_structured_configuration_variable_name: structured_configuration_variable_value {{ some_variable }}
+```
+
+At runtime, `eos_designs` searches for custom templates in several locations, therefore, custom templates should be saved in one of these locations:
+
+```text
+<path to users AVD implementation>/playbooks/templates/templates/<template name>
+<path to users AVD implementation>/playbooks/templates/<template name>
+<path to users AVD implementation>/playbooks/<template name>
+```
+
+The "templates" folders do not exist by default so will need to be created if that is the location chosen by the user.
+
+### Variable Validation
+
+As with any AVD variable, additional structured configuration variables will be subject to validation against the schema. Because they are custom variables, additional structured configuration variables will not be included in the existing schema and will therefore produce warning messages. To bypass validation, custom variables should start with an underscore.
```

Comment:
```suggestion
As with any AVD variable, additional structured configuration variables will be subject to validation against the schema. Because they are custom variables, additional structured configuration variables will not be included in the existing schema and will produce warning messages. Custom variables should start with an underscore if you would like to bypass validation.
```